src/components/selectionBoxLayer.ts

```diff
@@ -83,11 +81,53 @@ export module Components {
         topLeft: topLeft,
         bottomRight: bottomRight
       };
-      this._bindBoxDataValues();
+    }
+
+    // sets pixelPosition() or value() based on the other, depending on which was the last one set
+    private _syncPixelPositionAndValue() {
+      if (this._xScale != null) {
+        if (this._xBoundsMode === PropertyMode.VALUE && this._xExtent[0] != null
+            && this._xExtent[1] != null) {
+          let topLeft = {
+            x: this._xScale.scale(this._xExtent[0]),
+            y: this.bounds().topLeft.y
+          };
+          let bottomRight = {
+            x: this._xScale.scale(this._xExtent[1]),
+            y: this.bounds().bottomRight.y
+          };
+          this._setBounds({
+            topLeft: topLeft,
+            bottomRight: bottomRight
+          });
+        } else if (this._xBoundsMode === PropertyMode.PIXEL) {
+          this._xExtent = [this._xScale.invert(this.bounds().topLeft.x), this._xScale.invert(this.bounds().bottomRight.x)];
+        }
+      }
+      if (this._yScale != null) {
```

Comment:
Scenario (not sure if potential bug but):
- Set bounds to pixel equivalent of [a, b]
- Set yExtent to [c, d]
- Ask for bounds -> they will return pixel equivalent of [c, d]
- Remove yScale
- Ask for bounds -> they will return pixel equivalent of [a, b]